nemo/collections/asr/metrics/wer_bpe.py

```diff
@@ -12,18 +12,87 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from dataclasses import dataclass
 from typing import List
 
 import editdistance
 import torch
 from torchmetrics import Metric
 
-from nemo.collections.asr.metrics.wer import move_dimension_to_the_front
-from nemo.collections.asr.parts.utils.rnnt_utils import Hypothesis
+from nemo.collections.asr.metrics.wer import AbstractCTCDecoding, CTCCharDecodingConfig
 from nemo.collections.common.tokenizers.tokenizer_spec import TokenizerSpec
 from nemo.utils import logging
 
 
+class CTCBPEDecoding(AbstractCTCDecoding):
+    """
+    Used for performing CTC auto-regressive / non-auto-regressive decoding of the logprobs.
+
+    Args:
+        decoding_cfg: A dict-like object which contains the following key-value pairs.
+            strategy: str value which represents the type of decoding that can occur.
+                Possible values are :
+                -   greedy (for greedy decoding).
+
+            compute_timestamps: A bool flag, which determines whether to compute the character/subword, or
+                word based timestamp mapping the output log-probabilities to discrite intervals of timestamps.
+                The timestamps will be available in the returned Hypothesis.timestep as a dictionary.
+
+            ctc_timestamp_type: A str value, which represents the types of timestamps that should be calculated.
+                Can take the following values - "char" for character/subword time stamps, "word" for word level
+                time stamps and "all" (default), for both character level and word level time stamps.
+
+            word_seperator: Str token representing the seperator between words.
+
+            preserve_alignments: Bool flag which preserves the history of logprobs generated during
+                decoding (sample / batched). When set to true, the Hypothesis will contain
+                the non-null value for `logprobs` in it. Here, `logprobs` is a torch.Tensors.
+
+            batch_dim_index: Index of the batch dimension of ``targets`` and ``predictions`` parameters of
+                ``ctc_decoder_predictions_tensor`` methods. Can be either 0 or 1.
+
+            The config may further contain the following sub-dictionaries:
+            "greedy":
+                preserve_alignments: Same as above, overrides above value.
+                compute_timestamps: Same as above, overrides above value.
```

Comment:
Would it be desirable to keep WERBPE_TS and WER_TS separateley in decode_timestamp utils? If you think merging it into one place is more desirable, we can merge it later. Otherwise, we might keep it as usual.

Reply:
Seperately is better. Those two are doing some additional work that normal metric calculation isn't doing. Merging it will overly complicate the simpler metric calculation class.

Reply:
Got it.